src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -58,6 +58,13 @@ public CommonCompiler(CommandLineParser parser, string responseFile, string[] ar
             this.Arguments = parser.Parse(allArgs, baseDirectory, sdkDirectoryOpt, additionalReferenceDirectories);
             this.MessageProvider = parser.MessageProvider;
             this.AnalyzerLoader = analyzerLoader;
+
+#if DEBUG
+            if (Arguments.ParseOptions.Features.ContainsKey("debug-determinism"))
+            {
+                EmitDeterminismKey(Arguments, args, baseDirectory, parser);
+            }
+#endif
```

Comment:
I think it is fine to have this as an "unsupported" feature flag on all (not just Debug) builds. If it becomes popular or is demonstrated useful to external customers, we can consider exposing it as a supported option, perhaps modified or enhanced per feedback.